src/detect-engine-mpm.c

```diff
@@ -230,13 +230,15 @@ void DetectMpmInitializeAppMpms(DetectEngineCtx *de_ctx)
         /* default to whatever the global setting is */
         int shared = (de_ctx->sgh_mpm_ctx_cnf == ENGINE_SGH_MPM_FACTORY_CONTEXT_SINGLE);
 
+#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
```

Comment:
is this so slow? If so, can we optimize it instead of special casing it?

Reply:
fuzz_sigpcpap_aware, over its latest corpus, spends 5% of its time in `DetectMpmInitializeAppMpms`, and 28% of it in the `ConfGetBool` call + 8% in the confstring computation
This `#ifdef` looked like an easy win...

In Suricata use case, you are not supposed to reload rules every millisecond...
So, I guess this is not so slow for Suricata's use case.
But this is called many times by this fuzz target...

If you really want to optimize it, I guess `ConfNodeLookupChild` should not use a linked list and compare the strings but a more efficient structure like a hash table or some tree/dictionary...

But if you do not find this `#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION` an easy win, I would just forget it for now...